src/py42/clients/detectionlists/high_risk_employee.py

```diff
@@ -0,0 +1,140 @@
+import json
+
+from py42.clients import BaseClient
+from py42.clients.util import get_all_pages
+
+
+class HighRiskEmployeeClient(BaseClient):
+    """Administrator utility to manage High Risk employees.
+
+    `Support Documentation <https://support.code42.com/Administrator/Cloud/Monitoring_and_managing/Detection_list_management_APIs>`__
+
+    """
+
+    _api_version = u"v2"
+    _uri_prefix = u"/svc/api/{0}".format(_api_version)
+    _resource = u"/highriskemployee"
+
+    def __init__(self, session, user_context, user_client):
+        super(HighRiskEmployeeClient, self).__init__(session)
+        self._user_context = user_context
+        self._user_client = user_client
+
+    def _make_uri(self, action):
+        return u"{0}{1}{2}".format(self._uri_prefix, self._resource, action)
+
+    def _add_high_risk_employee(self, tenant_id, user_id):
+
+        data = {"tenantId": tenant_id, "userId": user_id}
+        uri = self._make_uri(u"/add")
+        return self._session.post(uri, data=json.dumps(data))
+
+    def add(self, user_id):
+        """Adds a user to high risk employee detection list.
+
+        A user can only be added to high risk employee detection list if a user profile
+        exists in detection list.
+
+        It would return failure when
+            - a user already exists in high risk employee detection list
+            - user_id is not found in detection lists
+
+        Args:
+            user_id (str or int): Id of the user who needs to be added to HRE detection list.
+
+        Returns:
+            :class:`py42.sdk.response.Py42Response`
+        """
+        if self._user_client.get_by_id(user_id):
```

Comment:
Let's bring in `py42.clients.UserClient` as a dependency and use it to determine the id's username if it has not already been already added to `DetectionLists`. Once have the username, we can call `DectectionUserClient.create()`, then call `_add_high_risk_employee()` as normal.

Reply:
Done, do note, used `user.client`.`get_by_uid` instead of `get_by_id` as `user_id` in detection list refers to `uid` of `user_client`

Reply:
Works for me!

Reply:
TIL